caps/capabilities.go

```diff
@@ -103,6 +106,27 @@ func (r *Repository) Add(cap *Capability) error {
 	return nil
 }
 
+// AddType adds a capability type to the repository.
+// Type names must be unique within the repository.  An error is returned if
+// this constraint is violated. Note that unlike capabilities, types are append
+// only. We don't have any reasonable scenarios where a capability type would
```

Comment:
I see nothing wrong with removing capability types, so I'd suggest not being prescriptive about what we don't know yet.

Suggested replacement for the whole comment:

// AddType adds a capability type to the repository.
// It's an error to add the same capability type more than once.